x-pack/metricbeat/module/aws/aws_test.go

```diff
@@ -167,6 +168,26 @@ func TestCheckTagFiltersExist(t *testing.T) {
 			},
 			false,
 		},
+		{
+			"more than one value per key of tagFilters is included in resourcegroupstaggingapi tags",
+			[]Tag{
+				{
+					Key:   "Organization",
+					Value: []string{"Product", "Engineering"},
+				},
+			},
+			[]resourcegroupstaggingapitypes.Tag{
```

Comment:
adding this makes the test fails:
```
				{
					Key:   awssdk.String(tagKey3),
					Value: awssdk.String(tagValue1),
				},
```

if we refactor [this](https://github.com/elastic/beats/pull/32775/commits/83dd4331f156af972c54d79197449e20c282ff4c#diff-e64e32ecae8005a2e832c30a2207ee1c8b2547336399f8a40db9a16a256e06ffR250-R260) to:
```
	existingTags := 0
	for _, tagFilter := range tagsFilter {
		for idx, tagKey := range tagKeys {
			if tagFilter.Key != tagKey {
				continue
			}
			valueExists, _ := StringInSlice(tagValues[idx], tagFilter.Value)
			if valueExists {
				existingTags++
			}
		}
	}
	return existingTags == len(tagsFilter)
```
we can even introduce `AND` and `OR` logic:
```
  tags_filter:
    - key: "same_key"
      value: ["AND"]
    - key: "same_key"
      value: ["LOGIC"]
```

vs

```
  tags_filter:
    - key: "same_key"
      value: ["OR", "LOGIC"]
```

what do you think, @cjr125 ?

(ps: please, do not force push on the branch if possible, it makes it difficult to track changes between commits, thanks :))

Reply:
Won't AND logic on a single key always fail by definition? will that be useful? AFAIK, we will always want AND across keys and OR within a key for real use cases.

Reply:
yes, sorry, you cannot have multiple keys with the same value, I didn't consider this

can you explain what is the logic applied to the above test case and why it should fail in case of adding `tagKey3` with `tagValue1`?

and what should be the logic applied to the following filter:
```
  tags_filter:
    - key: "key1"
      value: ["value1"]
    - key: "key2"
      value: ["value2"]
```

thanks :)

Reply:
The method checks to see if a value which is not specified in the list of tag values for a given key exists and rejects the record if so. Adding tagKey3 with tagValue1 does not represent a realistic use case because it would mean that a key was set to both a valid and invalid value on the same record.

As for your next example, the logic should be key1 == value1 && key2 == value2

Also, as a fun aside, I realized we can hack the current logic to get OR functionality by specifying an additional "module" block in the config :)

Reply:
thank you also for your help. i believe i have now addressed this feedback. also sorry for the force push but i did separate out a new commit (had to force push to sign it properly after i forgot the first time). let me know if any further enhancements are required.